### DISCUSSION TITLE
Filtering Budibase DB columns with spaces

### DISCUSSION THREAD

**Discussion by theetrain** [Budibase/budibase#6400] [Help]
Hi, I'm not sure whether to report this as a bug or not since I cannot tell if it's intended behaviour.

When filtering a Budibase DB table within the 'data' view, I'm unable to filter a column by value when that column contains spaces; the same issue occurs when using a data provider for that table. Should this be possible, or should I ensure all column names do not contain spaces?

This may relate to https://github.com/Budibase/budibase/issues/5715

**Comment by melohagan** [comment]
As suggested in the issue, try using square brackets for column names with spaces.

**Reply by theetrain** [reply]
Thanks, though the suggestion you pointed out ([surrounding values with brackets in handlebars expressions](https://github.com/Budibase/budibase/issues/5715#issuecomment-1130053444)) only applies when calling values using handlebars. In my case, I want to filter data using a static value, but the _column_ contains spaces. From what I can gather, here are the cases:

* When a column name contains 1 or 0 spaces, it is filterable in data providers and the 'Data' screen
* When a column name contains 2 or more spaces, it is NOT filterable

See the below screenshot that illustrates my findings. In this test page, every table is wrapped with individual data providers, and the headings describe what filters are applied.

On a somewhat-unrelated note, whenever I update the filters for my data providers, they would not save. My workaround is to delete my filters, and then add all-new filters in order to make changes to the data providers.

![Screenshot of test page illustrating how some data providers containing columns spaces may or may not be filterable](https://user-images.githubusercontent.com/12798751/174852863-2c8d3cf6-a1e6-4878-8d78-ccfa7b140459.png)

**Reply by melohagan** [reply]
Wow that's a very peculiar bug! I was able to recreate, so I've raised a bug here: https://github.com/Budibase/budibase/issues/6421

Also thank you very much for the detailed image!
